modules/account-lib/src/coin/hbar/transactionBuilder.ts

```diff
@@ -75,6 +110,47 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
   // endregion
 
   // region Common builder methods
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo A hedera memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    // strings are UTF-16, max 100 bytes
+    if (memo.length * 2 > 100) {
```

Comment:
You should turn this into a buffer/hex 1st and then check the length, no?

Reply:
Sure, we will convert it to buffer and check the length. This implementation was based on the SDK current implementation but we will change this and perform some surround testing to check if it keeps working properly.